### PR TITLE
Tox file for testing

### PR DESCRIPTION
**Pull request by ogenstad**
Adding tox.ini file as an example for [the discussion](https://github.com/napalm-automation/napalm/issues/327).

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[tox]
+envlist = py27,py35
+
+[testenv]
+deps =
+    -rrequirements-dev.txt
+
+commands=
+   py.test --cov-report= --cov=napalm_eos test/
+   pylama napalm_eos test
```
Comment (dbarrosop) [inline tox.ini]: shouldn't just be `pylama .`?

Comment (dbarrosop) [reply tox.ini]: Yeah, please, put `pylama .` back and add this to `setup.cfg` in the `pylama` section; `skip = .tox/*`. That should do.

**Comment by dbarrosop** [comment]
I like it, not sure if we need something more here, do you know if this is gonna have some impact on coveralls.

Anyway, we can probably test. If you do the changes I mentioned in the code, add `.tox` to gitignore if it's not there already, and update `.travis.yml` we can see how it behaves.

Btw, what's the implication on travis? Now we tell travis it has to run on different python version, do we have to just leave one and let travis handle it?

**Comment by dbarrosop** [comment]
Googling it seems the correct way is using; `tox-travis`. Do you mind playing with it? : )

**Comment by ogenstad** [comment]
I think the use of tox-travis is mostly to show the correct Python versions on Travis instead of showing the tox environment version of Python. Compare [this](https://travis-ci.org/napalm-automation/napalm-eos/builds/184164810) to [this](https://travis-ci.org/networklore/nelsnmp/builds/153480049), might be other things that I'm not aware of. Anyway I used tox-travis for this.

I also changed setup.cfg to have pytest call pylama instead of running pylama. The reason for this is that the tests will fail as soon after the first command has a non normal exit. So pylama will never run if the other tests are failing. This way linting errors will be shown along with the results from pytest.

**Comment by dbarrosop** [comment]
Works for me, let's see what the rest thinks.

**Comment by mirceaulinic** [comment]
Same here.

**Comment by dbarrosop** [comment]
I am merging this one and creating issues to do this on all the repos.

Thanks!

**Pull request merged**